components/dashboard/src/hooks/use-check-dedicated-setup.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License.AGPL.txt in the project root for license information.
+ */
+
+import { useFeatureFlags } from "../contexts/FeatureFlagContext";
+import { useAuthProviders } from "../data/auth-providers/auth-provider-query";
+import { useOrganizations } from "../data/organizations/orgs-query";
+import { useQueryParams } from "./use-query-param";
+
+const FORCE_SETUP_PARAM = "dedicated-setup";
+const FORCE_SETUP_PARAM_VALUE = "force";
+
+export const useCheckDedicatedSetup = () => {
+    const orgs = useOrganizations();
+    const authProviders = useAuthProviders();
+    const params = useQueryParams();
+    const { enableDedicatedOnboardingFlow } = useFeatureFlags();
+
+    const forceSetup = params.get(FORCE_SETUP_PARAM) === FORCE_SETUP_PARAM_VALUE;
+
+    const hasOrgs = (orgs.data || [])?.length > 0;
+    const hasAuthProviders = (authProviders.data || []).length > 0;
+
+    // Show setup if:
+    // - query param set to force
+    // - or
+    // - feature flag is on and either no orgs or no auth providers
+    return {
+        needsOnboarding: forceSetup || (enableDedicatedOnboardingFlow && (!hasOrgs || !hasAuthProviders)),
```

Comment:
I'm not sure if this is the logic we'll be able to go with, need to make some progress to understand a few things, namely:

* Will the admin user be added to the org that's created?
* After we have an active SSO config, will we return that as an auth provider from `getAuthProviders()`?

Reply:
I'm not sure, if we need to rely on (old) `authProviders` at all here.

Let's check the conditions where `needsOnboarding` should be true:
1.  `forceSetup` is set ✔️ 
2. guarded by `enableDedicatedOnboardingFlow`:
  * current account has no Org with SSO created
  * even if there is an Org, but no SSO created yet, we might be in re-entering state
 
It should be safe to assume for now, that on Dedicated there wouldn't be any `authProviders` returned. And on gitpod.io, this is always true. Which means, we can remove this from the equation.

Reply:
I've updated this to just use the query param check for now so it doesn't have any unexpected side-effects while we build more of the flow out. That will unblock dev and let us add the right backend check for if setup is needed. Then I can incorporate that call into this flow as well.